### PR TITLE
TRT-1305: Move Sippy to using variant registry

### PR DESCRIPTION
**Pull request by stbenjam**
Note: tests will fail until we get bigquery working in tests.  This change also means we always require bigquery.

- The OCP variant manager now queries bigquery
- Adds a new "sync-variants" command that will sync all jobs variants instead of waiting until we see them again (test this locally against a dev backup with `./sippy load --loader sync-variants --google-service-account-credential-file ~/tokens/asdf.json`)
- UI for tests shows the top 8 variants (PANTFUSI) with the rest in a tooltip

Example:
```
----------------------
name     | periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-heavy-build-ovn-remote-libvirt-s390x
variants | {Platform:libvirt,Architecture:s390x,Network:ovn,NetworkStack:ipv4,Topology:ha,FeatureSet:default,Upgrade:none,SecurityMode:default,Installer:ipi,Aggregation:none,NetworkAccess:default,Owner:eng,Release:4.16,ReleaseMajor:4,ReleaseMinor:16,Scheduler:default,Suite:unknown}
-[ RECORD 2 ]---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
name     | periodic-ci-openshift-multiarch-master-nightly-4.16-upgrade-from-nightly-4.15-ocp-e2e-upgrade-gcp-ovn-heterogeneous
variants | {Platform:gcp,Architecture:heterogeneous,Network:ovn,NetworkStack:ipv4,Topology:ha,FeatureSet:default,Upgrade:minor,SecurityMode:default,Installer:ipi,Aggregation:none,FromRelease:4.15,FromReleaseMajor:4,FromReleaseMinor:15,NetworkAccess:default,Owner:eng,Release:4.16,ReleaseMajor:4,ReleaseMinor:16,Scheduler:default,Suite:unknown}
-[ RECORD 3 ]---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
name     | periodic-ci-openshift-assisted-installer-agent-master-edge-subsystem-test-periodic
variants | {Architecture:amd64,NetworkStack:ipv4,Topology:ha,FeatureSet:default,Upgrade:none,SecurityMode:default,Installer:assisted,Aggregation:none,NetworkAccess:default,Owner:eng,Scheduler:default,Suite:unknown}
```

image

### CODE REVIEW

**Comment by stbenjam** [comment]
/hold

There's dependent changes I need to make in openshift/api before this can land.

File: pkg/testidentification/ocp_variants.go [hunk 1]
```diff
@@ -21,344 +25,154 @@ import (
...
-		variants = append(variants, "proxy")
-	}
+func (v *openshiftVariants) AllPlatforms() sets.String {
+	return v.variantValues["Platform"]
+}
 
-	if len(variants) == 0 {
-		log.Infof("unknown variant for job: %s\n", jobName)
-		return []string{"unknown variant"}
+func (v *openshiftVariants) IdentifyVariants(jobName, _ string, _ models.ClusterData) []string {
```
Comment (dgoodwin) [inline pkg/testidentification/ocp_variants.go:121]: Looks like you can kill off the release and cluster data args.

File: pkg/dataloader/variantsyncer/variantsyncer.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func (vl *VariantSyncer) Load() {
+	allJobs := loadAllProwJobs(vl.dbc)
+	for _, j := range allJobs {
+		newVariants := vl.mgr.IdentifyVariants(j.Name, j.Release, models.ClusterData{})
+		if !reflect.DeepEqual(newVariants, []string(j.Variants)) {
+			j.Variants = newVariants
+		}
+		if res := vl.dbc.DB.WithContext(context.TODO()).Save(j); res.Error != nil {
```
Comment (dgoodwin) [inline pkg/dataloader/variantsyncer/variantsyncer.go:48]: Should that jump up into the DeepEqual if? Can't recall if that's a no-op in gorm if nothing has changed.

File: pkg/testidentification/ocp_variants.go [hunk 3]
```diff
@@ -21,344 +25,154 @@ import (
...
-		"ovirt",
-		"vsphere-ipi",
-		"vsphere-upi",
-	)
+const (
+	NeverStable = "never-stable"
+
+	jobVariantsQuery = `SELECT
+  job_name,
+  ARRAY_AGG(STRUCT(variant_name, variant_value)) AS variants
```
Comment (dgoodwin) [inline pkg/testidentification/ocp_variants.go:45]: Cool, I need to learn how to use this more.

**Review by dgoodwin — COMMENTED** [review]
Looks great to me, would like to spend some time testing later.

**Comment by dgoodwin** [comment]
This is really neat. I did some local testing and everything seems to be working fine. The install/upgrade analysis pages become a little more interesting I think, they could however use a little UI tweaking, the column headers are a bit too wide now, I would suggest splitting into two lines on the colon. Also we could omit a few of them like ReleaseMajor/Minors to tighten it up a little, and it would be nice to get All to the front of the list, Aggregated bumped it. These could all come later though, it's functional enough as is.

We'll need to update our alerts a little once this goes in, anything else likely to break? 

Really glad you went full steam ahead on this instead of mimicking the old variants as I had planned, this is much better.

**Comment by dgoodwin** [comment]
/lgtm
/hold

Release when you're ready to merge. Could be a little rocky when it goes in.

**Comment by stbenjam** [comment]
/hold

Actually let's wait until after stand-up

**Pull request merged**